src/app/shared/product-filter/product-filter.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Product} from '../products/product.interface';
+
+@Pipe({
+    name: 'productFilter',
+})
+export class ProductFilterPipe implements PipeTransform {
+    transform(products: Product[] | null, name: string): Product | null {
+        if (!products) {
+            return null;
+        }
+
+        return products.find(product => product.name.includes(name)) || null;
```

Comment:
В `|| null` нет необходимости т.к. выше уже была проверка через if и выходит, что `products.find(...)` всегда будет возвращать пустой массив и при преобразовании в `boolean` будет всегда возвращать `true`. Получается, что сейчас `products.find(...) || null` эквивалентно `[...] || null` и в плане булевой проверки эквивалентно `true || null`)

```suggestion
        return products.find(product => product.name.includes(name));
```